File: slack-reader/src/test/java/dev/number6/slackreader/core/WorkspaceMessagesTest.java
package dev.number6.slackreader.core;

import dev.number6.slackreader.model.WorkspaceMessages;
import dev.number6.slackreader.generate.SlackReaderRDG;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import uk.org.fyodor.range.Range;

import java.util.ArrayList;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

class WorkspaceMessagesTest {

    private WorkspaceMessages testee;

    @BeforeEach
    void setup() {
        testee = new WorkspaceMessages();
    }

    @Test
    void activeChannelNamesReturnsChannelsWithMessages() {
        String inactiveChannelName = "inactive";

        testee.add(SlackReaderRDG.string(Range.closed(5, 30)).next(), SlackReaderRDG.list(SlackReaderRDG.string(), Range.closed(10, 20)).next());
        testee.add(inactiveChannelName, new ArrayList<>());
        testee.add(SlackReaderRDG.string(Range.closed(5, 30)).next(), SlackReaderRDG.list(SlackReaderRDG.string(), Range.closed(10, 20)).next());

        Collection<String> activeChannelNames = testee.getActiveChannelNames();

        assertThat(activeChannelNames).hasSize(2);
        assertThat(activeChannelNames).doesNotContain(inactiveChannelName);
    }

}

File: slack-reader/src/test/kotlin/dev/number6/slackreader/core/WorkspaceMessagesTest.kt
package dev.number6.slackreader.core

import assertk.assertThat
import assertk.assertions.doesNotContain
import assertk.assertions.hasSize
import dev.number6.slackreader.model.WorkspaceMessages
import org.junit.jupiter.api.Test
import uk.org.fyodor.generators.RDG
import uk.org.fyodor.range.Range
import java.util.*

internal class WorkspaceMessagesTest {

    @Test
    fun activeChannelNamesReturnsChannelsWithMessages() {
        val testee = WorkspaceMessages()
        val inactiveChannelName = "inactive"
        testee.add(RDG.string(Range.closed(5, 30)).next(), RDG.list(RDG.string(), Range.closed(10, 20)).next())
        testee.add(inactiveChannelName, ArrayList())
        testee.add(RDG.string(Range.closed(5, 30)).next(), RDG.list(RDG.string(), Range.closed(10, 20)).next())
        val activeChannelNames = testee.getActiveChannelNames()
        assertThat(activeChannelNames).hasSize(2)
        assertThat(activeChannelNames).doesNotContain(inactiveChannelName)
    }
}